packages/statemanager/src/stateManager.ts

```diff
@@ -962,21 +962,14 @@ export class DefaultStateManager implements EVMStateManagerInterface {
       throw new Error(`dumpStorage f() can only be called for an existing account`)
     }
     const trie = this._getStorageTrie(address, account)
+    const storage: StorageDump = {}
+    const stream = trie.createReadStream()
 
-    return new Promise((resolve, reject) => {
-      const storage: StorageDump = {}
-      const stream = trie.createReadStream()
+    for await (const chunk of stream) {
+      storage[bytesToHex(chunk.key)] = bytesToHex(chunk.value)
+    }
```

Comment:
I haven't looked too deeply, a general thing though:

We need to keep the API intact (aka: backwards compatible), so from `createReadStream()`, so the return type is therefore not allowed to change.

If I interpret the code above correctly (showing a usage example of `createReadStream()` it *does* here though (it does, right? so the new stream API is realized with async calls, the old one with events, if I see correctly?).

Just to clarify: I also like the new API a lot better.

We nevertheless need to find a way to remain compatibilty, otherwise things will break for users using the method in the old way.

Two eventual options:

1. Can we use the web stream API in the way the old one worked (just asking for completeness, I would otherwise assume: likely not)?
2. If not, we likely need to bite the sour apple and keep the old method "as is", keep the dependency for now, add a new method with an eventually temporary name (`createWebReadStream()`, `createAsyncReadStream()`,?) and then we add this whole thing to #3216 to remember to fully remove the dependency along the next breaking release round and eventually rename the new method name to the old one.

If we go 2. (guess that's likely?) please also add a `@deprecated` note to the code docs of the old method.

Reply:
Yes, it seems like the web API differs from `readable-stream` in that it uses async calls instead of events. I just took a deeper look and can confirm that there isn't an option to use events with the new web API, so I added the `@deprecated` note and made both options available to eventually update it in the future with a breaking release.

Reply:
👍